warehouse-management-service/internal/handler/handler.go

```diff
@@ -469,7 +479,226 @@ func (h *handler) DeleteShelfBlock(w http.ResponseWriter, r *http.Request) {
 			return
 		}
 	}
+}
+
+func (h *handler) GetShelf(w http.ResponseWriter, r *http.Request) {
+	shelfId := chi.URLParam(r, "shelfId")
+
+	if shelfId == "" {
+		err := fmt.Errorf("%v", api.ShelfResponse{
+			Error: "shelf id cannot be empty",
+		})
+		h.logger.Log(log.Error, err)
+		h.response(w, http.StatusBadRequest, err)
+		return
+	}
 
+	shelf, err := h.shelfService.GetShelfById(r.Context(), shelfId)
+	switch err {
```

Comment:
I understand that an error here can be 1 of 3 values. However, that still doesn't justify using a switch statement for that. I'd write this like because of 2 reasons:
1. Convention of error checking in Golang
2. Putting non-erroneous statements like http.StatusOK response in same indentation as the flow of program improves readability.

```golang
warehouse, err := h.warehouseService.GetWarehouseById(r.Context(), warehouseId)
	if err != nil {
		if err == wms.WarehouseDoesNotExist {
			h.logger.Log(log.Error, err)
			h.response(w, http.StatusNotFound, api.GetWarehouseResponse{Error: fmt.Sprintf(
				"failed to get, warehouse: %s does not exist",
				warehouseId,
			)})
		} else {
			h.logger.Log(log.Error, err)
			h.response(w, http.StatusInternalServerError, api.GetWarehouseResponse{Error: "Failed to get warehouse"})
		}

		return
	}

	h.response(w, http.StatusOK, api.GetWarehouseResponse{Response: *warehouse})
```

Reply:
Changed this to use `if err != nil` syntax

Reply:
I still see the `switch err` used in multiple places. Consider changing them to the `if` syntax. 
As I have elaborated above, switch, although convenient, tends to reduce readability, especially for the `nil` & `default` cases.

Reply:
Updated the syntax throughout the handler